Sources/StreamChatUI/ChatMessageList/MessageComposer/ChatMessageComposerVC.swift

```diff
@@ -343,63 +359,37 @@ open class _ChatMessageComposerVC<ExtraData: ExtraDataTypes>: _ViewController,
 
     // MARK: Attachments
 
-    public enum SelectedAttachments {
-        case media, documents
-    }
-    
-    open var selectedAttachments: SelectedAttachments? {
-        if imageAttachments.isEmpty, documentAttachments.isEmpty {
-            return .none
-        } else {
-            return imageAttachments.isEmpty ? .documents : .media
-        }
-    }
-    
-    open var imageAttachments: [URL] = [] {
-        didSet {
-            didUpdateImageAttachments()
-        }
-    }
-    
-    open var documentAttachments: [URL] = [] {
+    open var urlAttachments: [URL] = [] {
         didSet {
-            didUpdateDocumentAttachments()
+            didUpdateAttachments()
         }
     }
```

Comment:
Do we actually need this indirection? Why not use `var attachments: [AttachmentEnvelope]` directly? It will make it also possible to add custom attachments there by customers.

Reply:
My understanding is that we work with URLs to save resources until necessary, I don't see a problem with using `[AttachmentEnvelope]` directly other than it's consuming more memory and taking some CPU time. 
If I'm thinking about it, every change is directly visible in the Attachments view anyway, so in the end, it doesn't have to more expensive.

Reply:
Done